src/backends/oracle/standard-into-type.cpp

```diff
@@ -235,41 +235,48 @@ void oracle_standard_into_type_backend::pre_fetch()
 void oracle::read_from_lob(oracle_session_backend& session,
     OCILobLocator * lobp, std::string & value)
 {
-    ub4 len;
-
-    sword res = OCILobGetLength(session.svchp_, session.errhp_,
-        lobp, &len);
+    // We can't get the CLOB size in bytes directly, only in characters, which
+    // is useless for UTF-8 as it doesn't tell us how much memory do we
+    // actually need for storing it, so we'd have to allocate 4 bytes for every
+    // character which could be a huge overkill. So instead read the CLOB in
+    // chunks of its natural size until we get everything.
+    ub4 len = 0;
+    sword res = OCILobGetChunkSize(session.svchp_, session.errhp_, lobp, &len);
     if (res != OCI_SUCCESS)
     {
         throw_oracle_soci_error(res, session.errhp_);
     }
 
-    std::vector<char> buf(len);
+    value.clear();
```

Comment:
Could this be an issue if an exception is thrown later?

Reply:
I don't think we make any guarantees about what the contents of `value` is if an exception is thrown, so I don't think there is any issue here, but please let me know if I'm overlooking something.